src/Interpreters/ExternalLoader.cpp

```diff
@@ -818,13 +818,10 @@ class ExternalLoader::LoadingDispatcher : private boost::noncopyable
             if (!min_id)
                 min_id = getMinIDToFinishLoading(forced_to_reload);
 
-            if (info->state_id >= min_id)
-                return true; /// stop
```

Comment:
`info->loading_id <= info->state_id` always, so this condition became unnecessary.